### ISSUE TITLE
Batch Size Dependent Output Values of Spectral Loss and Cluster Loss in `torch_geometric.nn.dense.DMoNPooling`

### ISSUE THREAD

**Issue by wwang-chcn** [pyg-team/pytorch_geometric#8284]
### 🐛 Describe the bug

The output spectral_loss and cluster_loss from DMoNPooling exhibit discrepancies with the documented values. Specifically, the values of spectral_loss and cluster_loss are dependent on the batch size, as evidenced by the examples below:

```{Python}
import torch
import torch.nn as nn
from torch_geometric.nn.dense import DMoNPooling


# we define a graph with 4 node and 3 features
N, F = 4, 3
X = torch.Tensor([[100,-100,0]]).expand(N,F)
adj_0 = torch.ones(N)
adj_0[0] = 0
adj_1 = torch.zeros(N)
adj_1[0] = 1
adj = torch.cat((adj_0.reshape(1,N),adj_1.reshape(1,N).expand(N-1,N))).float().reshape(N,N)

print(f'X: {X}')
print(f'adj: {adj}')

# define a DMoN layer to cluter these nodes into 2 categories
C = 2
dmon = DMoNPooling(channels=F,k=C)
dmon.mlp.lins[0].weight = nn.Parameter(torch.Tensor([[1,0,0],[0,1,0]]))
dmon.mlp.lins[0].bias = nn.Parameter(torch.Tensor([0,0]))

# Observing the results when the batch size is set to 1
def dmon_batch_test(B = 1):
    B = B
    batch_X = X.unsqueeze(0)
    batch_adj = adj.unsqueeze(0)
    batch_X_ = batch_X.expand(B,N,F)  # duplicate with B times
    batch_adj_ = batch_adj.expand(B,N,N)  # duplicate with B times
    s, out, out_adj, spectral_loss, ortho_loss, cluster_loss = dmon(batch_X_,batch_adj_)
    print(f'batch size: {B}.')
    print(f'\tspectral_loss: {spectral_loss.item()}.')
    print(f'\tortho_loss: {ortho_loss.item()}.')
    print(f'\tcluster_loss: {cluster_loss.item()}.')


dmon_batch_test(B=1)
# batch size: 1.
#     spectral_loss: -0.25.
#     ortho_loss: 0.7653668522834778.
#     cluster_loss: 0.41421353816986084.


# Observing the results when the batch size is set to 2, 3, and 4 with the same graph information for each batch
dmon_batch_test(B=2)
# batch size: 2.
#     spectral_loss: -0.125.
#     ortho_loss: 0.7653668522834778.
#     cluster_loss: 0.9999998807907104.

dmon_batch_test(B=3)
# batch size: 3.
#     spectral_loss: -0.0833333358168602.
#     ortho_loss: 0.7653668522834778.
#     cluster_loss: 1.4494895935058594.

dmon_batch_test(B=4)
# batch size: 4.
#     spectral_loss: -0.0625.
#     ortho_loss: 0.7653668522834778.
#     cluster_loss: 1.8284270763397217.
```

As the batch size increases, spectral_loss decreases, while cluster_loss increases.

### Environment

Environment 1
* PyG version: 2.3.1
* PyTorch version: 2.2.0.dev20230922
* OS: MacOS
* Python version: 3.10.10
* CUDA/cuDNN version: N/A
* How you installed PyTorch and PyG (`conda`, `pip`, source): pip
* Any other relevant information (*e.g.*, version of `torch-scatter`):

Environment 2
* PyG version: 2.3.1
* PyTorch version: 2.0.1
* OS: Linux
* Python version: 3.10.10
* CUDA/cuDNN version: 11.8
* How you installed PyTorch and PyG (`conda`, `pip`, source): pip
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by wwang-chcn** [comment]
Another inconsistence with DMoN paper: Activation function of MLP should be None rathen SeLU.

**Comment by BozianuLeon** [comment]
I've also been looking into the DMoN paper and noticed that the trainable skip connection (eq 3) is not included in the implementation here. Is there a reason for this?

**Comment by wwang-chcn** [comment]
> I've also been looking into the DMoN paper and noticed that the trainable skip connection (eq 3) is not included in the implementation here. Is there a reason for this?

I don't think so. The trainable skip connection (eq 3) is an adjustment of GCN layers, while DMoN is only an implementation of node pooling.
In the authors' original implementation in TF, they also implement GCN and DMoN in two parts. One thing the authors forgot to mention in the paper (eq 4) is that there is an Linear (Dense) layer with trainable parameters following the GCN output.

**Comment by BozianuLeon** [comment]
Precisely, you're right. And in their TF implementation they also dont output the adjacency matrix, as is done here in PyG. Thanks for opening this issue, I'll follow the PR you've opened :)
